js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -2636,8 +2666,11 @@ module.exports = {
                     }
 
                     const modifiedValues = parameters.row && (parameters.row.isNewRow ? parameters.row.values : parameters.row.modifiedValues);
-
-                    if(modifiedValues && modifiedValues[columnIndex] !== undefined && parameters.column && !isCommandCell && parameters.column.setCellValue) {
+                    const isCellInvalidInNewRow = parameters.row && parameters.row.isNewRow && editingController.isInvalidCell({
+                        rowKey: parameters.key,
+                        columnIndex: parameters.column.index
+                    });
+                    if(((modifiedValues && modifiedValues[columnIndex] !== undefined) || isCellInvalidInNewRow) && parameters.column && !isCommandCell && parameters.column.setCellValue) {
```

Comment:
Create virtual method _isCellModified instead of isValidCell and move logic with calling isValidCell to it. Editing should never know about validation.

Reply:
OK